cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -52,212 +49,333 @@ func TestConstructor(t *testing.T) {
 	assert.NotNil(t, newConsumer)
 }
 
-// partitionConsumerWrapper wraps a Sarama partition consumer into a Sarama cluster partition consumer
-type partitionConsumerWrapper struct {
-	topic     string
-	partition int32
-
-	sarama.PartitionConsumer
+func TestSaramaConsumerWrapper_MarkPartitionOffset(t *testing.T) {
+	sc := &kmocks.Consumer{}
+	metadata := "meatbag"
+	sc.On("MarkPartitionOffset", topic, partition, msgOffset, metadata).Return()
+	sc.MarkPartitionOffset(topic, partition, msgOffset, metadata)
+	sc.AssertCalled(t, "MarkPartitionOffset", topic, partition, msgOffset, metadata)
 }
 
-func (s partitionConsumerWrapper) Partition() int32 {
-	return s.partition
+// NewTestConfig returns a config meant to be used by tests.
+// Due to inconsistencies with the request versions the clients send using the default Kafka version
+// and the response versions our mocks use, we default to the minimum Kafka version in most tests
+func NewTestConfig() *sarama.Config {
+	config := sarama.NewConfig()
+	config.Version = sarama.MinVersion
+	return config
 }
 
-func (s partitionConsumerWrapper) Topic() string {
-	return s.topic
+type Store struct {
+	store     *memory.Store
+	consumer  *Consumer
+	done      sync.WaitGroup
+	t         *testing.T
+	returnErr bool
 }
 
-func newSaramaClusterConsumer(saramaPartitionConsumer sarama.PartitionConsumer, mc *smocks.PartitionConsumer) *kmocks.Consumer {
-	pcha := make(chan cluster.PartitionConsumer, 1)
-	pcha <- &partitionConsumerWrapper{
-		topic:             topic,
-		partition:         partition,
-		PartitionConsumer: saramaPartitionConsumer,
+func (s *Store) WriteSpan(ctx context.Context, span *model.Span) error {
+	err := s.store.WriteSpan(ctx, span)
+	s.t.Log("Write span store!!!")
+	go func() {
+		s.consumer.Close()
+		s.done.Done()
+	}()
+	if err == nil && s.returnErr {
+		err = errors.New("Write span store failed")
```

Comment:
```suggestion
		err = errors.New("write span store failed")
```

Reply:
This extra forced error handling is invalid, now use `require. NoError(s.t, err)`